sdk/test-utils/test-utils/src/tracing/azureTracing.ts

```diff
@@ -0,0 +1,127 @@
+import { OperationTracingOptions, useInstrumenter } from "@azure/core-tracing";
+import { assert } from "chai";
+import { TestInstrumenter } from "./testInstrumenter";
+import { SpanGraph, SpanGraphNode } from "./spanGraphModel";
+// this is the plugin used in the test file
+function chaiAzureTrace(chai: Chai.ChaiStatic, _utils: Chai.ChaiUtils): void {
+  // expect(() => {}).to.supportsTracing() syntax
+  chai.Assertion.addMethod("supportsTracing", function(
+    expectedSpanNames: string[],
+    options?: any,
+    thisArg?: any
+  ) {
+    return supportsTracing(this._obj, expectedSpanNames, options, thisArg);
+  });
+  // assert.supportsTracing(() => {}) syntax
+  chai.assert.supportsTracing = supportsTracing;
+}
+
+// Implementation for supportsTracing Plugin
+async function supportsTracing<
+  Options extends { tracingOptions?: OperationTracingOptions },
+  Callback extends (options: Options) => unknown
+>(
+  callback: Callback,
+  expectedSpanNames: string[],
+  options?: Options,
+  thisArg?: ThisParameterType<Callback>
+) {
+  const instrumenter = new TestInstrumenter();
+  useInstrumenter(instrumenter);
+  const startSpanOptions = {
+    packageName: "test",
+    ...options
+  };
+  const { span: rootSpan, tracingContext } = instrumenter.startSpan("root", startSpanOptions);
+
+  const newOptions = {
+    ...options,
+    tracingOptions: {
+      tracingContext: tracingContext
+    }
+  } as Options;
+  await callback.call(thisArg, newOptions);
+  console.log(callback.name);
```

Comment:
We should remove all the console.log statements to avoid noise when running tests